src/hooks/useTxDetails/useTxDetails.ts

```diff
@@ -25,21 +35,37 @@ export interface TxDetails {
   precision: number
   explorerTxLink: string
   explorerAddressLink: string
+  direction: 'in-place' | 'outbound' | 'inbound'
 }
 
 export const getStandardTx = (tx: Tx) => (tx.transfers.length === 1 ? tx.transfers[0] : undefined)
-export const getBuyTx = (tx: Tx) =>
-  !!tx.tradeDetails ? tx.transfers.find(t => t.type === chainAdapters.TxType.Receive) : undefined
-export const getSellTx = (tx: Tx) =>
-  !!tx.tradeDetails ? tx.transfers.find(t => t.type === chainAdapters.TxType.Send) : undefined
+export const getBuyTx = (tx: Tx) => tx.transfers.find(t => t.type === chainAdapters.TxType.Receive)
+export const getSellTx = (tx: Tx) => tx.transfers.find(t => t.type === chainAdapters.TxType.Send)
 
 export const useTxDetails = (txId: string, activeAsset?: Asset): TxDetails => {
   const tx = useSelector((state: ReduxState) => selectTxById(state, txId))
+  const method = tx.data?.method
+
+  const isSupportedContract = method ? SUPPORTED_CONTRACT_METHODS.has(method) : false
```

Comment:
this little dude could move out above

Reply:
Moved out in https://github.com/shapeshift/web/pull/1076/commits/066dedd1f599ad8d3f4f26daf2a2b06de1e097a6.